elytron/ELY-2538-caching-realm-update-cached-credential.adoc

```diff
@@ -0,0 +1,80 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/ELY-2538[ELY-2538]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-2069[EAP7-2069] - EAP issue tracker
+* https://issues.redhat.com/browse/ELY-2538[ELY-2538] - Elytron project issue tracker
+* https://issues.redhat.com/browse/WFLY-18185[WFLY-18185] - Wildfly community documentation issue tracker
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Krepinska]
+
+=== QE Contacts
+
+* mailto:TODO
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron project]
+* Wildfly Security component will be consequently affected by the improvements introduced by this new feature
+
+== Requirements
+
+=== Hard Requirements
+
+* It must be possible for the caching realm to authenticate a user and update the cached credential if it has been changed externally and a user attempts to authenticate with the new credential.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Implementation Plan
+
+This functionality should either be added as a default, as this was the behaviour of the legacy security subsystem. Or we can add this behaviour under a new attribute of the caching realm. If we add it as configurable with a property, then it can be backported with a system property.
+
+I have decided to enable this functionality by default. The reason for this is that this functionality was present in the legacy security, and users can consider the absence of this functionality to be a bug, so we should not require its configuration.
+
+Specification of the functionality that will be added: if a user currently cached by the caching realm has had their credentials updated externally, then if this user provides an updated password when authenticating to the WildFly server, the caching realm will be able to successfully authenticate a user. This is because if the caching realm receives a credential that is different from the credential that the identity has cached, it will attempt to authenticate directly with the underlying realm. If the authentication succeeds, the credential in the cache will be updated. Outdated credential will therefore not be possible to be used anymore.
+
+== Test Plan
+
+* WildFly Elytron test suite: Functional tests
+
+== Community Documentation
+
+WildFly - community documentation will be delivered in form of a PR to a main branch
+
+== Release Note Content
+
+It is now possible to update cached credentials with new credentials and authenticate to the WildFly server if those credentials have been updated outside of WildFly in the user store.
```

Comment:
You mention "it is now possible" but if this is now the default behaviour it should probably be stated as that.

Reply:
Updated